appserver/ejb/ejb-opentracing/src/main/java/fish/payara/ejb/opentracing/OpenTracingIiopInterceptorFactory.java

```diff
@@ -0,0 +1,99 @@
+ * exception as provided by the Payara Foundation in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.ejb.opentracing;
+
+import fish.payara.opentracing.OpenTracingService;
+import org.glassfish.enterprise.iiop.api.IIOPInterceptorFactory;
+import org.glassfish.hk2.api.ServiceLocator;
+import org.glassfish.internal.api.Globals;
+import org.jvnet.hk2.annotations.Service;
+import org.omg.IOP.Codec;
+import org.omg.PortableInterceptor.ClientRequestInterceptor;
+import org.omg.PortableInterceptor.ORBInitInfo;
+import org.omg.PortableInterceptor.ServerRequestInterceptor;
+
+import javax.inject.Singleton;
+
+/**
+ * Factory for creating IIOP client and server interceptors that propagate OpenTracing SpanContext.
+ *
+ * @author Andrew Pielage <andrew.pielage@payara.fish>
+ */
+@Service(name = "OpenTracingIiopInterceptorFactory")
+@Singleton
+public class OpenTracingIiopInterceptorFactory implements IIOPInterceptorFactory {
+
+    public static final int OPENTRACING_IIOP_ID = 3226428;
+    public static final long OPENTRACING_IIOP_SERIAL_VERSION_UID = 20200731171822L;
+
+
+    private ClientRequestInterceptor clientRequestInterceptor;
+    private ServerRequestInterceptor serverRequestInterceptor;
+
+    @Override
+    public ClientRequestInterceptor createClientRequestInterceptor(ORBInitInfo info, Codec codec) {
+        ServiceLocator staticBaseServiceLocator = Globals.getStaticBaseServiceLocator();
+        if (clientRequestInterceptor == null) {
+            OpenTracingService openTracingService = staticBaseServiceLocator.getService(OpenTracingService.class);
+
+            if (openTracingService != null) {
+                clientRequestInterceptor = new OpenTracingIiopClientInterceptor(openTracingService);
+            }
+        }
+
+        return clientRequestInterceptor;
+    }
+
+    @Override
+    public ServerRequestInterceptor createServerRequestInterceptor(ORBInitInfo info, Codec codec) {
```

Comment:
This and the above method basically do the same just for different types. Could be extracted.